components/autofill/src/sync/credit_card/mod.rs

```diff
@@ -3,4 +3,113 @@
 * file, You can obtain one at http://mozilla.org/MPL/2.0/.
 */
 
-pub mod engine;
+pub mod incoming;
+
+use super::engine::{ConfigSyncEngine, EngineConfig, SyncEngineStorageImpl};
+use super::{MergeResult, Metadata, ProcessIncomingRecordImpl, SyncRecord};
+use crate::db::models::credit_card::InternalCreditCard;
+use crate::error::*;
+use crate::sync_merge_field_check;
+use incoming::CreditCardsImpl;
+use rusqlite::Transaction;
+use std::sync::{Arc, Mutex};
+use sync_guid::Guid as SyncGuid;
+use types::Timestamp;
+
+// The engine.
+pub fn create_engine(
+    db: Arc<Mutex<crate::db::AutofillDb>>,
+) -> ConfigSyncEngine<InternalCreditCard> {
+    ConfigSyncEngine {
+        db,
+        config: EngineConfig {
+            namespace: "credit_cards".to_string(),
+            collection: "creditcards",
+        },
+        storage_impl: Box::new(CreditCardsEngineStorageImpl {}),
+    }
+}
+
+pub(super) struct CreditCardsEngineStorageImpl {}
+
+impl SyncEngineStorageImpl<InternalCreditCard> for CreditCardsEngineStorageImpl {
+    fn get_incoming_impl(&self) -> Box<dyn ProcessIncomingRecordImpl<Record = InternalCreditCard>> {
```

Comment:
This is not a useful comment because I'm not quite sure what to suggest as an improvement, however . . . at a high level I understand what's happening in the `get_incoming_impl` function but I fear the uninitiated (or even future me) wouldn't.